substrate/frame/support/src/traits/voting.rs

```diff
@@ -59,16 +59,16 @@ pub trait VoteTally<Votes, Class> {
 	/// users.
 	fn setup(class: Class, granularity: Perbill);
 }
-pub enum PollStatus<Tally, Moment, Class> {
+pub enum PollStatus<Tally, Moment, Class, Origin> {
 	None,
-	Ongoing(Tally, Class),
+	Ongoing(Tally, Class, Origin),
 	Completed(Moment, bool),
 }
 
-impl<Tally, Moment, Class> PollStatus<Tally, Moment, Class> {
-	pub fn ensure_ongoing(self) -> Option<(Tally, Class)> {
+impl<Tally, Moment, Class, Origin> PollStatus<Tally, Moment, Class, Origin> {
```

Comment:
Actually thinking about this again, maybe we should introduce some `DispatchablePollStatus: PollStatus` that only gives you the origin. 

There can be polls that are not dispatchables and then you would not have an `Origin`.